jobrunner/executors/volumes.py

```diff
@@ -154,7 +160,18 @@ def find_newer_files(job, reference):
         return found
 
 
-def get_volume_api():
+def default_volume_api():
     module_name, cls = config.LOCAL_VOLUME_API.split(":", 1)
     module = importlib.import_module(module_name)
     return getattr(module, cls)
+
+
+DEFAULT_VOLUME_API = default_volume_api()
+
+
+def get_volume_api(job):
+    for api in [BindMountVolumeAPI, DockerVolumeAPI]:
+        if api.volume_exists(job):
+            return api
+
+    return DEFAULT_VOLUME_API
```

Comment:
Clever! So in theory we can just switch the default, new jobs start using the new API and any existing jobs continue with the old one.

Reply:
Zactly